src/lib.rs

```diff
@@ -885,6 +894,13 @@ impl Node {
 
 		log_info!(self.logger, "Connected to peer {}@{}. ", peer_info.node_id, peer_info.address);
 
+		let is_announcement_broadcast_started =
```

Comment:
I am thinking something like
```rust
tokio::task::block_in_place(move || {
	runtime.block_on(async move {
		if !is_initial_connection_established {
			self.is_initial_connection_established.store(true, Ordering::Release);
		}
		con_cm.connect_peer_if_necessary(con_node_id, con_addr).await
	})
}
```
I am not so confident about this because I am setting the state before we connect to the node

**OR**

```rust
tokio::task::block_in_place(move || {
	runtime.block_on(async move {
		con_cm.connect_peer_if_necessary(con_node_id, con_addr).await
	})
}
if !is_initial_connection_established {
	self.is_initial_connection_established.store(true, Ordering::Release);
}
```

I am guessing here that `block_in_place` will wait till the async call ends

Reply:
Can you review the new changes?, I hope got this correct this time